### ISSUE TITLE
Syntax error at or near ")"

### ISSUE THREAD

**Issue by filipmaljkovic** [emaprojects/ecoreDiagnostics#2]
I am able to successfully run the code from CodeToRun.R, except for the last command. Since I have only one cohort in the database, as you can see:
![ecorediagnostics](https://user-images.githubusercontent.com/77625787/117131369-a4bb2480-ada1-11eb-887d-b91a007bd2a7.png)
I run the following code:
`CohortDiagnostics::launchCohortExplorer(connectionDetails = connectionDetails, cdmDatabaseSchema = cdmDatabaseSchema, cohortDatabaseSchema = cohortDatabaseSchema, cohortTable = cohortTable, cohortId = 4470)`
But I get the following error:
```
DBMS:
postgresql

Error:
org.postgresql.util.PSQLException: ERROR: syntax error at or near ")"
  Position: 132

SQL:
SELECT  subject_id
  FROM (
    SELECT DISTINCT subject_id 
    FROM results.ecoreDiagnostics 
    WHERE cohort_definition_id = 
  ) all_ids
  ORDER BY MD5(RANDOM()::TEXT || CLOCK_TIMESTAMP()::TEXT) LIMIT 100;

R version:
R version 4.0.3 (2020-10-10)

Platform:
x86_64-w64-mingw32

Attached base packages:
- stats
- graphics
- grDevices
- utils
- datasets
- methods
- base

Other attached packages:
- gdtools (0.2.3)
- magrittr (2.0.1)
- shiny (1.6.0)
- ecoreDiagnostics (0.0.1)
- DatabaseConnector (4.0.2)
```
It seems like the parameter value for cohortId doesn't get relayed to the code, or am I missing something?

**Comment by emaprojects** [comment]
It looks to me exactly as you suggest: cohortId isn't being parsed into the paremeterised Sql query.

The functionality is part of the main [CohortDiagnostics](https://github.com/OHDSI/CohortDiagnostics) package and I must admit that I am not familiar with the _Cohort Explorer_ which I think is relatively new.

I had a quick dig and it looks like the Sql query in question is being parameterised here:

https://github.com/OHDSI/CohortDiagnostics/blob/755eedbccd73b909197718f5b31f8ce5bc6e16c7/inst/shiny/CohortExplorer/global.R#L22-L33

It's looking for `shinySettings$cohortDefinitionId` but when I look to see how `shinySettings` is populated inside `launchCohortExplorer` the parameter is named `cohortId` rather then `cohortDefinitionId`:

https://github.com/OHDSI/CohortDiagnostics/blob/755eedbccd73b909197718f5b31f8ce5bc6e16c7/R/Shiny.R#L206-L212

Overall I'm a little confused by this as I can't see how `launchCohortExplorer` would ever work in its current form. Unless you can see something that I'm missing, I think it should be raised as an issue inside CohortDiagnostics.

Out of interest, are you able to run `launchDiagnosticsExplorer` OK?

Thanks so much for running the code so far and for your effort looking into the error!

**Comment by filipmaljkovic** [comment]
Yes, I was able to run `launchDiagnosticsExplorer` without any problems.

I'm even less familiarized with CohortDiagnostics, so can't be of much help. I just need to run these. :)

**Comment by hms1** [comment]
Understood :) 

As long as `launchDiagnosticsExplorer` works OK, then I think you can consider the package run complete - looks to me like a bug in `launchCohortExplorer`.

Thanks again!

**Comment by filipmaljkovic** [comment]
Cool, thanks for the explanation :)
